Lib/importlib/_bootstrap_external.py

```diff
@@ -1635,3 +1635,29 @@ def _install(_bootstrap_module):
     supported_loaders = _get_supported_file_loaders()
     sys.path_hooks.extend([FileFinder.path_hook(*supported_loaders)])
     sys.meta_path.append(PathFinder)
+
+
+def alternative_architectures(suffixes):
+    """Add a suffix with an alternative architecture name
+    to the list of suffixes so an extension built with
+    the default (upstream) setting is loadable with our Pythons
+    """
+
+    arch_map = {
+        "arm-linux-gnueabi": "arm-linux-gnueabihf",
+        "armeb-linux-gnueabi": "armeb-linux-gnueabihf",
+        "mips64-linux-gnu": "mips64-linux-gnuabi64",
+        "mips64el-linux-gnu": "mips64el-linux-gnuabi64",
+        "ppc-linux-gnu": "powerpc-linux-gnu",
+        "ppc-linux-gnuspe": "powerpc-linux-gnuspe",
+        "ppc64-linux-gnu": "powerpc64-linux-gnu",
+        "ppc64le-linux-gnu": "powerpc64le-linux-gnu",
+    }
+
+    for suffix in suffixes:
+        for original, alternative in arch_map.items():
+            if original in suffix:
+                suffixes.append(suffix.replace(original, alternative))
+                return suffixes
```

Comment:
About this return: We know there is only one suffix here that needs to add the alternative?

Reply:
Yes, there should be only one main suffix containing SOABI. See: https://github.com/fedora-python/cpython/blob/fedora-3.9/Python/dynload_shlib.c#L37

There is also `ALT_SOABI` but it has no impact on `EXT_SUFFIX` which is used when building extensions.

Reply:
And `ALT_SOABI` is defined only `if test "$Py_DEBUG" = 'true' -a "$with_trace_refs" != "yes";`